meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -211,51 +178,52 @@ function fillTabletData(clinicObj) {
     });
 }
 
-//This method will handle all the data related to week columns "tot and reg"
-function buildWeekData(clinicObj, minuWeek, colKey) {
-    var weekVal = week - minuWeek;
-    $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
-        $.each(tabletDataArray, function(index, value) {
-            if (value.DevicesIDs === "- DeviceId  " + deviceVal.deviceId) {
-                var itemName = colKey + weekVal;
-                value[itemName] = deviceVal.submissionsCount;
-            }
-        });
-    });
-}
 
+//this method will handle to fill the data from the apis into the table
+function fillAPIResultData(clinicObj, minusNum, colKey) {
 
-//This method will handle all the data related to year columns
-function buildYearData(clinicObj, minuYear, colKey) {
-    var yearVal = current_Year - minuYear;
-    $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
-        $.each(tabletDataArray, function(index, value) {
-            if (value.DevicesIDs === "- DeviceId  " + deviceVal.deviceId) {
-                var itemName = colKey + yearVal;
-                value[itemName] = deviceVal.submissionsCount;
-            }
+    //This section will handle all the data related to week columns "tot and reg"
```

Comment:
Comments should be more to the point in my opinion. I'd go with:
`data related to week columns "tot and reg"`
Moreover some `if`s have comments before the `if` statement (like in this case) while others inside the block (e.g. line 197). You should be consistent.
In my opinion what's going on in those if&elses is very clear just from conditions.
E.g.
`colKey == "casesInWeek" || colKey == "noOfRegistersInWeek"` is much more readable for me then:
`//This section will handle all the data related to week columns "tot and reg"`
I suggest removing all of the comments from lines: 185, 197 and 208.

Reply:
comments removed